engine/client/client.go

```diff
@@ -126,6 +131,7 @@ func Connect(ctx context.Context, params Params) (_ *Client, _ context.Context,
 	c.internalCtx, c.internalCancel = context.WithCancel(context.WithoutCancel(ctx))
 	c.closeCtx, c.closeRequests = context.WithCancel(context.WithoutCancel(ctx))
 
+	// FIXME: is this still needed? (not a problem, just want to be sure)
 	c.eg, c.internalCtx = errgroup.WithContext(c.internalCtx)
```

Comment:
The errgroup is still useful for running the session attachables server async, so it's fine for now I think. The ctxs in this file are still confusing though, but would leave as is for now.